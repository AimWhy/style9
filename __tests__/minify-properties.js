/* eslint-env jest */
const compile = require('./compile.js');

it('does not minify by default', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    opacity: 1
  }
});
  `;
  const { code } = compile(input);

  expect(code).toMatchSnapshot();
});

it('does not minify styles', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    opacity: 1
  }
});
  `;
  const { styles } = compile(input);

  expect(styles).toMatchSnapshot();
});

it('minifies known properties', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    opacity: 1
  }
});
  `;
  const { code } = compile(input, { minifyProperties: true });

  expect(code).toMatchSnapshot();
});

it('hashes unknown properties', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    foo: 'bar'
  }
});
  `;
  const { code } = compile(input, { minifyProperties: true });

  expect(code).toMatchSnapshot();
});

it('minifies nested properties', () => {
  const input = `
import style9 from 'style9';
const styles = style9.create({
  default: {
    '@media (max-width: 1000px)': {
      '::before': {
        opacity: 1
      }
    }
  }
});
  `;
  const { code } = compile(input, { minifyProperties: true });

  expect(code).toMatchSnapshot();
});
