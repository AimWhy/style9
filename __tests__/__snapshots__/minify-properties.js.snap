// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not minify by default 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    opacity: \\"cu2kwdz\\"
  }
};
styles.default;"
`;

exports[`does not minify styles 1`] = `".cu2kwdz{opacity:1}"`;

exports[`hashes unknown properties 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    \\"1wfoylc\\": \\"cxrs67x\\"
  }
};
styles.default;"
`;

exports[`minifies known properties 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    j9: \\"cu2kwdz\\"
  }
};
styles.default;"
`;

exports[`minifies nested properties 1`] = `
"import style9 from 'style9';
const styles = {
  default: {
    tp4qmw: {
      \\"1jy3mw3\\": {
        j9: \\"cog47hh\\"
      }
    }
  }
};
styles.default;"
`;
